multicluster/controllers/multicluster/clusterset_controller.go

```diff
@@ -18,38 +18,93 @@ package multicluster
+//       member and a leader, 2 instances of the controller will need to be run, one as a leader
+//		 the second as a member.
 type ClusterSetReconciler struct {
 	client.Client
-	Scheme *runtime.Scheme
+	mutex    sync.Mutex
+	Scheme   *runtime.Scheme
+	IsLeader bool
+
+	clusterSetConfig *multiclusterv1alpha1.ClusterSet
+	clusterSetID     common.ClusterSetID
+	clusterID        common.ClusterID
+
+	// These fields are only applicable on leader cluster
+	LocalClusterManager core.LocalClusterManager
+
+	// These fields are only applicable on member cluster
+	RemoteClusterManager core.RemoteClusterManager
 }
 
 //+kubebuilder:rbac:groups=multicluster.crd.antrea.io,resources=clustersets,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups=multicluster.crd.antrea.io,resources=clustersets/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups=multicluster.crd.antrea.io,resources=clustersets/finalizers,verbs=update
 
-// Reconcile is part of the main kubernetes reconciliation loop which aims to
-// move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the ClusterSet object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
+// Reconcile ClusterSet CRD changes
 func (r *ClusterSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	clusterSet := &multiclusterv1alpha1.ClusterSet{}
+	err := r.Get(ctx, req.NamespacedName, clusterSet)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return ctrl.Result{}, err
+		}
+		// if errors.IsNotFound(err)
+		klog.Infof("Received ClusterSet delete", "config", clusterSet, "leader", r.IsLeader)
+		if !r.IsLeader {
+			if err := r.RemoteClusterManager.Stop(); err != nil {
+				return ctrl.Result{}, err
+			}
+		} else {
+			if err := r.LocalClusterManager.Stop(); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+		return ctrl.Result{}, nil
+	}
+
+	klog.Infof("Received ClusterSet add/update", "config", clusterSet, "leader", r.IsLeader)
 
-	// your logic here
+	// Handle create or update
+	// If create
+	//   if leader, make sure the local cluster claim is part of the leader config
+	//   if not leader, make sure the local cluster claim is part of the member config
+	if err = r.validateLocalClusterClaim(clusterSet); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// if update,
+	//    if leader - nothing to do in inbound mode
+	//    if member - if leaders have changed handle accordingly, else nothing to do.
+	if !r.IsLeader {
+		err = r.updateMultiClusterSetOnMemberCluster(clusterSet)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
```

Comment:
here also need to create a localclustermanager `r.createOrUpdateMultiClusterSetOnLeaderCluster(clusterSet)`, otherwise, remote_resourceimport_controller will fail due to no localclustermanager, I would suggest to rename the method `createOrUpdateMultiClusterSetOnLeaderCluster` to `createOrUpdateMultiClusterSetLocalManager`

Reply:
Done, I ported your fix and saw this.